.travis.yml

```diff
@@ -1,36 +1,38 @@
 sudo: required
 services:
   - docker
-language: bash
+language: python
 env:
   global:
+    - IMAGE_NAME=praekeltfoundation/vumi
     - REGISTRY_USER=praekeltorgdeploy
     - secure: "I3t1OHRkMqYALcImjqlfldP9348bMgRAaD/sNy6aZ+4PPGVPurlGHU3jMEQWJJ9ojOcAAoN1qC+8kVsM1Fj5r9jmLUm1ixFE83gCbesD0EzZ5nPR4DMi8YR6Fde2nJcPgUoOOiNtE6/ZXHtJT3F4R8xbandsCGjae9jcHHGwSkIBoS8A6iUnOX5SE6sLrSydAvVAPzrfxaogKDrWdrWyBLLd3Un/nSK+v2ATg3b01W4cU/Wbz591Q2M4Bi2Y2cdETHzs/higertEFFKEHOIr2PvD9AfJWcJp+JjHxpu17pkgPxwLmmf/ORfo56udV+wxkDFWA1HjOYWSvUpHpkOBVY6hWmZ1AiB6nqA0iqTix/dpC9ppFLxRaNNv9xZMbbUiNqjk/P5BVmwc41RGM1Ye43uD2+yMah600Z44bSipQvhBUiv0Nr1igzYeb7kUWZq572SN7V0BSSO+JL0wneHcRFjvAVYRWYkoIdoIvW9CPt2pRQ56mvmHsqiMQE+2r3x0IEzGO44fEFL7AH4y82TrGdyYbLc8VL90QQiligovwpL89SndCbdrm3ThhKv+9coay+WVtfyk5SQ1kK+9kQU23hkI5dMM5dwNrOLxxhllyXcaE8ii79/gvotY2AKt3rFvWJWfk+DGm87unSq73dn5eQqW5Mhoezg+PvlSxrHMxx4="
   matrix:
-    - BASE_OS=alpine VARIETY=     TAG_DEFAULT=
-    - BASE_OS=debian VARIETY=     TAG_DEFAULT=1
-    - BASE_OS=debian VARIETY=pypy TAG_DEFAULT=
+    - VARIANT=
+    - VARIANT=alpine
+    - VARIANT=pypy
 
-# Update Docker Engine
-before_install:
-  - sudo apt-get update
-  - sudo apt-get install -qy -o Dpkg::Options::="--force-confold" docker-engine
+install: [] # Don't do a pip install (default for Travis Python environment)
 
 before_script:
-  - IMAGE_TAG="praekeltfoundation/vumi:$BASE_OS${VARIETY:+-$VARIETY}"
-  - DOCKERFILE="$BASE_OS/${VARIETY:-Dockerfile}${VARIETY:+.dockerfile}"
+  - image_tag="$IMAGE_NAME${VARIANT:+:$VARIANT}"
+  - dockerfile="${VARIANT:+$VARIANT/}Dockerfile"
+  - version="$(sed -nE 's/^vumi==([0-9\.]+)/\1/p' requirements.txt)"
+  - echo "Building image '$image_tag' from Dockerfile '$dockerfile' for Vumi version '$version'..."
+  - docker pull "$image_tag" || true
 
 script:
-  - docker build --tag "$IMAGE_TAG" --file "$DOCKERFILE" .
-  - ./test.sh "$IMAGE_TAG"
+  - docker build --pull --cache-from "$image_tag" --tag "$image_tag" --file "$dockerfile" .
+  - ./test.sh "$image_tag"
 
 after_script:
   - docker images
 
 before_deploy:
+  - pip install docker-ci-deploy==0.3.0
   - docker login -u "$REGISTRY_USER" -p "$REGISTRY_PASS"
 deploy:
   provider: script
-  script: ./deploy.sh "$IMAGE_TAG" ${TAG_DEFAULT:+--default}
+  script: dcd --version "$version" --version-latest "$image_tag"
   on:
-    branch: develop
+    branch: master
```

Comment:
I'm slowly moving most of these `docker-` repos to use `master` as the default branch. The production images come from this branch and `develop` is not the best name for that.